kernel/src/engine/sync/fs_client.rs

```diff
@@ -97,20 +97,53 @@ impl FileSystemClient for SyncFilesystemClient {
 
 #[cfg(test)]
 mod tests {
-    use std::fs::File;
     use std::io::Write;
+    use std::os::unix::fs::MetadataExt;
+    use std::{fs::File, time::UNIX_EPOCH};
 
     use bytes::{BufMut, BytesMut};
+    use itertools::Itertools;
     use url::Url;
 
     use super::SyncFilesystemClient;
-    use crate::FileSystemClient;
+    use crate::{FileMeta, FileSystemClient};
 
     /// generate json filenames that follow the spec (numbered padded to 20 chars)
     fn get_json_filename(index: usize) -> String {
         format!("{index:020}.json")
     }
 
+    #[test]
+    fn test_file_meta_is_correct() -> Result<(), Box<dyn std::error::Error>> {
+        let client = SyncFilesystemClient;
+        let tmp_dir = tempfile::tempdir().unwrap();
+
+        let path = tmp_dir.path().join(get_json_filename(1));
+        let mut f = File::create(path)?;
+        writeln!(f, "null")?;
+
+        let url_path = tmp_dir.path().join(get_json_filename(1));
+        let url = Url::from_file_path(url_path).unwrap();
+        let list: Vec<_> = client.list_from(&url)?.try_collect()?;
+
+        let metadata = f.metadata()?;
+        // We assert that the timestamp is in milliseconds
+        let expected_timestamp = metadata
+            .modified()?
+            .duration_since(UNIX_EPOCH)?
+            .as_millis()
+            .try_into()?;
```

Comment:
This is the exact same code the iterator uses... so it's not actually testing for correctness but rather testing that the code is equivalent. Not sure what the better test would be? Maybe we write a file and manually set its mtime, then verify the FileMeta has the same timestamp? 

(same criticism applies to the default client test)

Reply:
maybe create the file then just assert that the timestamp we get back is within 10s or 60s or something?

Reply:
Implemented a test that checks that the filemeta is from the last minute.